src/components/content/credential/AddCredential.tsx

```diff
@@ -0,0 +1,248 @@
+                form.setFieldsValue({ variables: credentialVariableList });
+            } else {
+                setCredentialVariableList([]);
+                form.setFieldsValue({ variables: [] });
+            }
+        });
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [currentCsp, currentType]);
+
+    function setVariablesValue(index: number, e: ChangeEvent<HTMLInputElement>) {
+        credentialVariableList[index].value = e.target.value;
+        form.setFieldsValue({ variables: credentialVariableList });
+    }
+
+    const columns: ColumnsType<CredentialVariable> = [
+        {
+            title: 'name',
+            dataIndex: 'name',
+        },
+        {
+            title: 'value',
+            dataIndex: 'value',
+            render: (value, record, index) => (
+                <Form.Item
+                    name='value'
+                    rules={[
+                        {
+                            required: isContainsEmpty(credentialVariableList),
+                            message: 'mandatory field',
+                        },
+                    ]}
+                >
+                    {''}
+                    {record.isSensitive ? (
+                        <Input.Password
+                            onChange={(e) => {
+                                setVariablesValue(index, e);
+                            }}
+                        />
+                    ) : (
+                        <Input
+                            onChange={(e) => {
+                                setVariablesValue(index, e);
+                            }}
+                        />
+                    )}
+                </Form.Item>
+            ),
+        },
+        {
+            title: 'description',
+            dataIndex: 'description',
+        },
+    ];
+
+    const isContainsEmpty = (credentialVariableList: CredentialVariable[]) => {
+        if (credentialVariableList.length === 0) {
+            return false;
+        }
+        return credentialVariableList.some(
+            (credentialVariable) => credentialVariable.isMandatory && !credentialVariable.value
+        );
+    };
+
+    const submit = (credentialVariables: CredentialVariables) => {
+        if (!isContainsEmpty(credentialVariables.variables)) {
+            const createCredential: CreateCredential = credentialVariables as CreateCredential;
+            const userName: string | null = localStorage.getItem(usernameKey);
+            if (userName === null) {
+                return;
+            }
+            createCredential.xpanseUser = userName;
+            void CredentialsManagementService.addCredential(createCredential)
+                .then(() => {
+                    getCredentials();
+                    getTipInfo('success', 'Add Credential Success!');
+                    onReset();
+                    onCancel();
+                })
+                .catch(() => {
```

Comment:
This error handling is not working when Xpanse API returns HTTP 400 with responses like the one in below. The GUI shows no error. 

{
   "resultType":"Credential Capability Not Found",
   "details":[
      "Defined credentials with type VARIABLES and name dfdfdf provided by csp VARIABLES not found."
   ],
   "success":false
}

Sample error handling can be found here. The exact error mentioned in `detailed` list must be displayed. 
This logic must be added to add, update, delete

https://github.com/eclipse-xpanse/xpanse-ui/blob/1447b9c25b960a22c6d7352c000b5358165dc2b4/src/components/content/order/OrderSubmitFailed.tsx#L19
https://github.com/eclipse-xpanse/xpanse-ui/blob/1447b9c25b960a22c6d7352c000b5358165dc2b4/src/components/content/register/RegisterResult.tsx#L44